tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -1102,96 +1102,94 @@ Status FragmentMetadata::store_v15_or_higher(
   auto num = num_dims_and_attrs();
   uint64_t offset = 0, nbytes;
 
+  // provide for error clean up on early exits.
+  ScopedExecutor premature_exit_cleanup([this]() { clean_up(); });
```

Comment:
You don't need to use `ScopedExecutor` here. It's sufficient to do three things:
* Put all the code that calls fallible functions `store_*` into a single `try` block.
* Change `RETURN_NOT_OK_ELSE` to `throw_if_not_ok`.
* Put the call to `clean_up` in the `catch` block.
This has an additional advantage that you can then also change the return type to `void` and let it throw exceptions instead of returning `Status`. The statement `throw` after calling `clean_up()` accomplishes this.